runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricUrns.java

```diff
@@ -29,14 +30,17 @@
    *
    * <p>Should be consistent with {@code parse_namespace_and_name} in monitoring_infos.py.
    */
+  @Nullable
   public static MetricName parseUrn(String urn) {
-    if (urn.startsWith(USER_COUNTER_URN_PREFIX)) {
-      urn = urn.substring(USER_COUNTER_URN_PREFIX.length());
+    if (urn.startsWith(USER_METRIC_URN_PREFIX)) {
+      urn = urn.substring(USER_METRIC_URN_PREFIX.length());
+    } else {
+      return null;
```

Comment:
Did you encounter this somewhere? Hopefully this isn't called if the metric isn't actually starting with USER_METRIC_URN_PREFIX

Reply:
As of this PR, this helper is called in two places:
- [`MonitoringInfoMetricName.parseUrn`](https://github.com/apache/beam/pull/7868/files#diff-ba1b936a7d4c8d43789e81a816b9bb90R55):
  - use the parsed user-`MetricName`, if possible
  - `throw` if it's not a user-metric (signaled by `null` return here)
- [`FlinkMetricContainer.updateMetrics`](https://github.com/apache/beam/pull/7868/files#diff-bd906385f3e3ff74094985af80da5a41R108):
  - use the parsed user-`MetricName`, if possible
  - skip the metric if it's not a user-metric (again, signaled by `null` return)
    - in later PRs I make Flink support system-metrics as well.

This was the most direct way to support those uses. The Flink metrics do actually see system metrics, and should gracefully skip them instead of `throw`ing (for now).